packages/cmr-client/src/EarthdataLogin.ts

```diff
@@ -0,0 +1,202 @@
+ *
+ * @example
+ * const { EarthdataLogin } = require('@cumulus/cmr-client');
+ *
+ * const earthdataLogin = new EarthdataLogin({
+ *  "username": "my-username",
+ *  "password": "my-password",
+ *  "edlEnv": "my-edl-environment"
+ * });
+ */
+export class EarthdataLogin {
+  username: string;
+  password: string;
+  edlEnv: string;
+
+  /**
+  * The constructor for the EarthdataLogin class
+  *
+  * @param {string} params.username - Earthdata Login username, required parameter
+  * @param {string} params.password - Earthdata Login password, required parameter
+  * @param {string} params.edlEnv - the Earthdata Login environment (['PROD', 'OPS', 'SIT', 'UAT'])
+  *
+  * @example
+  *
+  * {
+  *  "username": "janedoe",
+  *  "password": "password",
+  *  "edlEnv": "UAT"
+  * }
+  */
+
+  constructor(params: EarthdataLoginParams) {
+    this.username = params.username;
+    this.password = params.password;
+    this.edlEnv = params.edlEnv;
+  }
+
+  /**
+   * The method for getting the Earthdata Login endpoint URL based on the EDL environment
+   *
+   * @returns {Promise.<string>} the endpoint URL
+   */
+  getEDLurl(
+  ) {
+    switch (this.edlEnv) {
+      case 'PROD':
+      case 'OPS':
+        return 'https://urs.earthdata.nasa.gov';
+      case 'UAT':
+        return 'https://uat.urs.earthdata.nasa.gov';
+      case 'SIT':
+      default:
+        return 'https://sit.urs.earthdata.nasa.gov';
+    }
+  }
+
+  /**
+   * The method that determines if a user has a token, to call the retrieveEDLToken function
+   * which makes an API call to the Earthdata Login endpoint, or to create the token with the
+   * createEDLToken function. Returns the token as a string.
+   *
+   * @returns {Promise.<string>} the token
+   */
+  async getEDLToken(): Promise<string> {
+    let token = await this.retrieveEDLToken();
+    if (token === undefined) {
+      token = await this.createEDLToken();
+    }
+    return token;
+  }
+
+  /**
+   * The method for getting the token from the Earthdata Login endpoint. Sends a GET request
+   * with the users' base64 encoded username and password as a header for authorization. If the
+   * users' credentials are accepted a token is retrieved, if one exists, and returned on their
+   * behalf, if not, an error is thrown. If the user does not have a token in Earthdata Login
+   * then undefined is returned to indicate to token needs to be created.
+   *
+   * @returns {Promise.<string>} the token
+   */
+  async retrieveEDLToken(): Promise<string> {
+    const buff = Buffer.from(`${this.username + ':' + this.password}`).toString('base64');
+    const url = this.getEDLurl();
+    // response: get a token from the Earthdata login endpoint using credentials if exists
+    let response: EarthdataGetTokenResponse;
+    try {
+      response = await got.get(`${url}/api/users/tokens`,
+        {
+          headers: {
+            Authorization: `Basic ${buff}`,
+          },
+        }).json();
+    } catch (error) {
```

Comment:
To reduce the code redundancy of these methods, suggest to add a new method to handle the error (line 124-129)